api/v3/OfflineAutoRenewalJob.php

```diff
@@ -7,10 +7,18 @@
  * @return array
  */
 function civicrm_api3_offline_auto_renewal_job_run($params) {
+  $lock = Civi::lockManager()->acquire('worker.membershipextras.offlineautorenewal');
+  if (!$lock->isAcquired()) {
+    return civicrm_api3_create_error('Could not acquire lock, another Offline Autorenewal process is running');
+  }
+
   $offlineAutoRenewalJob = new CRM_MembershipExtras_Job_OfflineAutoRenewal();
```

Comment:
just checking...

if for some reason, an exception is thrown here `offlineAutoRenewalJob->run()`, the lock will not be released.. another try-catch block here?

Reply:
good catch, updated the PR